Gems/ROS2/Code/Source/Odometry/ROS2OdometrySensorComponent.cpp

```diff
@@ -55,34 +53,28 @@ namespace ROS2
         required.push_back(AZ_CRC_CE("ROS2Frame"));
     }
 
-    // ROS2SensorComponent overrides ...
-    void ROS2OdometrySensorComponent::SetupRefreshLoop()
-    {
-        InstallPhysicalCallback();
-    }
-
-    void ROS2OdometrySensorComponent::OnPhysicsInitialization(AzPhysics::SceneHandle sceneHandle)
+    void ROS2OdometrySensorComponent::OnOdometryEvent(AzPhysics::SceneHandle sceneHandle)
     {
-        AzPhysics::RigidBody* rigidBody = nullptr;
-        AZ::EntityId entityId = GetEntityId();
-        Physics::RigidBodyRequestBus::EventResult(rigidBody, entityId, &Physics::RigidBodyRequests::GetRigidBody);
-        AZ_Assert(rigidBody, "Entity %s does not have rigid body.", entityId.ToString().c_str());
-
-        m_bodyHandle = rigidBody->m_bodyHandle;
-
-        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
-        auto* simulatedBodyPtr = sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, m_bodyHandle);
-        auto rigidbodyPtr = azrtti_cast<AzPhysics::RigidBody*>(simulatedBodyPtr);
-        AZ_Assert(rigidbodyPtr, "Requested simulated body is not a rigid body");
-        m_initialTransform = rigidbodyPtr->GetTransform();
-    }
+        if (m_bodyHandle == AzPhysics::InvalidSimulatedBodyHandle)
+        {
+            AzPhysics::RigidBody* rigidBody = nullptr;
+            AZ::EntityId entityId = GetEntityId();
+            Physics::RigidBodyRequestBus::EventResult(rigidBody, entityId, &Physics::RigidBodyRequests::GetRigidBody);
+            AZ_Assert(rigidBody, "Entity %s does not have rigid body.", entityId.ToString().c_str())
+
+            m_bodyHandle = rigidBody->m_bodyHandle;
+
+            auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+            auto* simulatedBodyPtr = sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, m_bodyHandle);
+            auto* rigidbodyPtr = azrtti_cast<AzPhysics::RigidBody*>(simulatedBodyPtr);
+            AZ_Assert(rigidbodyPtr, "Requested simulated body is not a rigid body")
+            m_initialTransform = rigidbodyPtr->GetTransform();
+        }
 
-    void ROS2OdometrySensorComponent::OnPhysicsSimulationFinished(AzPhysics::SceneHandle sceneHandle, float deltaTime)
-    {
         auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
```

Comment:
AZ_Assert(sceneInterface, "No sceneInterface")

Reply:
We talked about this for the previous PR - then, I assumed that scene interface is retrieved from engine and it will not return nullptr. Do you think, that this is necessary/usefull anyway?